internal/app/backend/backend_service_test.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend
+
+import (
+	"io"
+	"testing"
+
+	"open-match.dev/open-match/internal/app/frontend"
+
+	"github.com/rs/xid"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"open-match.dev/open-match/internal/app/mmlogic"
+	"open-match.dev/open-match/internal/pb"
+	"open-match.dev/open-match/internal/rpc"
+
+	rpcTesting "open-match.dev/open-match/internal/rpc/testing"
+	statestoreTesting "open-match.dev/open-match/internal/statestore/testing"
+)
+
+func TestAssignTicketsEmptyRequest(t *testing.T) {
+	assert := assert.New(t)
+	tc := createBackendForTest(t)
+	defer tc.Close()
+
+	assignTicketsLoop(tc, &pb.AssignTicketsRequest{}, func(_ *pb.AssignTicketsResponse, err error) {
```

Comment:
This is a good example of where you can create a struct that has a input / response - and iterate over it and call the same assign as different test cases.